### PR TITLE
add command login to arlas_cli confs

### PR DESCRIPTION
**Pull request by sylvaingaudan**
Add `arlas_cli confs login`. This creates a configuration for cloud.arlas.io, highly simplified compared to `arlas_cli confs create`: just need the login/pwd for arlas and elastic + elastic endpoint.

Also, the created configuration becomes the default configuration: this means that other commands do not need to have the `--config` argument.

```shell
> arlas_cli confs login me@gisaia.com gisaia.com_ingest https://...

Creating configuration for cloud.arlas.io.sylvain.gaudan ...
Using gisaia.com as your organisation name.
Please enter your password for ARLAS Cloud (account me@gisaia.com)
Password: 
Thank you, now, please enter your password for elasticsearch (account gisaia.com_ingest)
Password: 
Configuration cloud.arlas.io.me created.
cloud.arlas.io.me is now your default configuration.
```

Command line help:

```
 Usage: python -m arlas.cli.cli confs login [OPTIONS] AUTH_LOGIN ELASTIC_LOGIN                        
                                            ELASTIC                                                   
                                                                                                      
 Add a configuration for arlas cloud                                                                  
                                                                                                      
╭─ Arguments ────────────────────────────────────────────────────────────────────────────────────────╮
│ *    auth_login         TEXT  login [default: None] [required]                                     │
│ *    elastic_login      TEXT  elasticsearch login [default: None] [required]                       │
│ *    elastic            TEXT  elasticsearch url [default: None] [required]                         │
╰────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Options ──────────────────────────────────────────────────────────────────────────────────────────╮
│ --auth-org                                 TEXT  ARLAS IAM Organization, default is your email     │
│                                                  domain name                                       │
│                                                  [default: None]                                   │
│ --allow-delete        --no-allow-delete          Is delete command allowed for this configuration? │
│                                                  [default: allow-delete]                           │
│ --auth-password                            TEXT  password [default: None]                          │
│ --elastic-password                         TEXT  elasticsearch password [default: None]            │
│ --help                                           Show this message and exit.                       │
╰────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

### CODE REVIEW

**Review by MohamedHamouGisaia — APPROVED** [review]

File: arlas/cli/configurations.py [hunk 1]
```diff
@@ -27,7 +29,7 @@ def create_configuration(
     persistence: str = typer.Option(default=None, help="ARLAS Persistence url"),
     persistence_headers: list[str] = typer.Option([], help="header (name:value)"),
 
-    elastic: str = typer.Option(default=None, help="dictionary of name/es resources"),
+    elastic: str = typer.Option(default=None, help="elasticsearch url"),
```
Comment (QuCMGisaia) [inline arlas/cli/configurations.py:32]: Elasticsearch should be capitalized

File: arlas/cli/configurations.py [hunk 2]
```diff
@@ -74,6 +76,60 @@ def create_configuration(
     print("Configuration {} created.".format(name))
 
 
+@configurations.command(help="Add a configuration for arlas cloud", name="login")
```
Comment (QuCMGisaia) [inline arlas/cli/configurations.py:79]: ARLAS

File: arlas/cli/configurations.py [hunk 3]
```diff
@@ -74,6 +76,60 @@ def create_configuration(
...
+def login(
+    auth_login: str = typer.Argument(help="login"),
+    elastic_login: str = typer.Argument(help="elasticsearch login"),
+    elastic: str = typer.Argument(help="elasticsearch url"),
+    auth_org: str = typer.Option(default=None, help="ARLAS IAM Organization, default is your email domain name"),
+    allow_delete: bool = typer.Option(default=True, help="Is delete command allowed for this configuration?"),
+    auth_password: str = typer.Option(default=None, help="password"),
```
Comment (QuCMGisaia) [inline arlas/cli/configurations.py:86]: ARLAS password

File: arlas/cli/settings.py [hunk 4]
```diff
@@ -31,11 +31,26 @@ class Settings(BaseModel):
...
+    def solve_config(config: str):
+        if not config:
+            if Configuration.settings.default:
+                print("Using default configuration {}".format(Configuration.settings.default))
+                return Configuration.settings.default
+            else:
+                print("Error: No default configuration, please provide one among [{}]".format(", ".join(Configuration.settings.arlas.keys())), file=sys.stderr)
+                exit(1)
+        if Configuration.settings.arlas.get(config, None) is None:
+            print("Error: arlas configuration {} not found among [{}]".format(config, ", ".join(Configuration.settings.arlas.keys())), file=sys.stderr)
```
Comment (QuCMGisaia) [inline arlas/cli/settings.py:50]: ARLAS

**Review by QuCMGisaia — APPROVED** [review]

File: arlas/cli/configurations.py [hunk 5]
```diff
@@ -74,6 +76,60 @@ def create_configuration(
...
+    elastic_password: str = typer.Option(default=None, help="elasticsearch password")
+):
+    if not re.match(r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$', auth_login):
+        print("Error: login {} is not a valid email".format(auth_login), file=sys.stderr)
+        exit(1)
+
+    name = "cloud.arlas.io." + auth_login.split("@")[0]
```
[the remark this answers is not in the document]
Comment (sylvaingaudan) [reply arlas/cli/configurations.py:93]: I thought that it was what we agreed on but you said cloud.arlas.io.**myname** yesterday, so I imagined that we in fact did not agree on it.

**Review by mbarbet — APPROVED** [review]

**Pull request merged**